### DISCUSSION TITLE
We need to integrate wallet connect into a Flutter application (ios, android)?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1558] [Developer Support]
What would be the recommended way of doing this - wrapper around the ios/kotlin libraries or the flutter native one?

**Comment by RootSoft** [comment]
If you just want to focus on the existing WC SDKs (Android, iOS & Web), I'd highly recommend to implement a [federated plugin](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#federated-plugins) architecture as a [Plugin package](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#plugin) package. Most Saas platforms that offer Flutter SDKs work this way, e.g [Sentry](https://github.com/getsentry/sentry-dart/tree/main/flutter).

Federated plugins have several advantages over single-package plugins, including:

* A plugin author does not need to have domain expertise for every supported Flutter platform (Android, iOS, Web, Mac OS, etc.).
* You can add support for a new platform without the original plugin author needing to review and pull in your code.
* Each package can be maintained and tested separately.
* Updates can be brought out quicker by updating the native SDKs and writing the `MethodChannel`.

I'd only go with a dart-only approach if you also want to focus on Linux, MacOS & Windows without planning to write platform native code and have a full-time dev.

**Comment by mwawrusch** [comment]
Thanks for the advice. It seems that "There's multiple frameworks that don't integrate with SPM like Flutter" might be an issue

**Reply by mwawrusch** [reply]
Also looks like we need to use walletconnect_dart till sunset period for wallets ends.

**Reply by Zombieliu** [reply]
Personally looking forward to the launch of the library

**Comment by makinghappen** [comment]
would be great to have [walletconnect_dart](https://github.com/RootSoft/walletconnect-dart-sdk) upgraded to WallectConnect 2.0 because [wallet-connect-dart](https://github.com/Orange-Wallet/wallet-connect-dart) does not support session initiation and generation of wc: URI, it relies on the WalletConnect SDK and also does not support V2.

**Comment by chadyj** [comment]
Thanks for the feedback.
We have another Flutter thread over here https://github.com/WalletConnect/walletconnect-monorepo/discussions/862#discussioncomment-4290619

Closing this for now.
